### PR TITLE
Enable config of tfstate location & re-add --rm to docker command

### PR DESCRIPTION
**Pull request by nissessenap**
Solves #31, the user can now define
RG_LOCATION_LONG & RG_LOCATION_SHORT to overwrite the default tfstate location
westeurope & we.

* Remove some non-used code
* Re-add --rm to the docker command

### CODE REVIEW

**Review by simongottschlag — CHANGES_REQUESTED** [review]
I think this change could be much smaller. Pass the variables to the env file if and then use `VAR=${VAR:-default}` in the script.

Let us take a look at it together tomorrow!

**Comment by nissessenap** [comment]
I have verified that it works to update the value as suggested in the makefile, but you can also ignore it and use the default values.
I'm running plan on a existing env and I tried to change RG_LOCATION_LONG but I don't know how much terraform uses it so that's why I think it still works even if I write something like

```
RG_LOCATION_SHORT=we
RG_LOCATION_LONG=foo
```

**Review by bittrance — COMMENTED** [review]
I think it is somewhat confusing that some parameters are passed via command line and others are passed via environment variables. For me, the change to the readme above is illustrates why this approach is problematic. I think it is time for `terraform.sh` to grow proper `getopts`,

File: docker/terraform.sh [hunk 1]
```diff
@@ -7,15 +7,14 @@ ENVIRONMENT=$3
...
-RG_LOCATION_LONG="westeurope"
+RG_LOCATION_SHORT=${RG_LOCATION_SHORT:-we}
+RG_LOCATION_LONG=${RG_LOCATION_LONG:-westeurope}
 BACKEND_KEY="${ENVIRONMENT}.terraform.tfstate"
 BACKEND_RG="rg-${ENVIRONMENT}-${RG_LOCATION_SHORT}-${SUFFIX}"
 BACKEND_KV="kv-${ENVIRONMENT}-${RG_LOCATION_SHORT}-${SUFFIX}"
 BACKEND_KV_KEY="sops"
 BACKEND_NAME="sa${ENVIRONMENT}${RG_LOCATION_SHORT}${SUFFIX}"
 CONTAINER_NAME="tfstate-${DIR}"
-ENVIRONMENT_FILE="/tmp/${ENVIRONMENT}.env"
```
Comment (bittrance) [inline docker/terraform.sh:18]: :+1:

File: README.md [hunk 2]
```diff
@@ -30,7 +32,7 @@ $(error Need to set DIR)
 endif
 
 AZURE_DIR_MOUNT:=-v $(AZURE_CONFIG_DIR):/work/.azure
-DOCKER_RUN:=docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint /opt/terraform.sh --env-file $(TEMP_ENV_FILE) $(AZURE_DIR_MOUNT) -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE)
+DOCKER_RUN:=docker run --user $(shell id -u) $(TTY_OPTIONS) --rm --entrypoint /opt/terraform.sh --env-file $(TEMP_ENV_FILE) $(AZURE_DIR_MOUNT) -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE)
```
Comment (simongottschlag) [inline README.md:35]: Either a separate PR or at least update the PR description to explain why this is done.

File: README.md [hunk 3]
```diff
@@ -72,6 +74,9 @@ setup:
 		echo AWS_PAGER= >> $(TEMP_ENV_FILE)
 	fi
 
+	echo "RG_LOCATION_SHORT=$(RG_LOCATION_SHORT)" >> $(TEMP_ENV_FILE)
+	echo "RG_LOCATION_LONG=$(RG_LOCATION_LONG)" >> $(TEMP_ENV_FILE)
```
Comment (simongottschlag) [inline README.md]: Looking at the other configurations, we don't enclose the text in `"` on the other places.

**Review by simongottschlag — APPROVED** [review]
I've added a few comments that I would like to be fixed, after that feel free to merge! Good job! :)

**Pull request merged**